zwave_js_server/util/node.py

```diff
@@ -115,12 +74,160 @@ async def async_set_config_parameter(
             f"Must provide a value within the target range ({', '.join(bounds)})"
         )
 
+    return new_value
+
+
+def partial_param_bit_shift(property_key: int) -> int:
+    """Get the number of bits to shift the value for a given property key."""
+    # We can get the binary representation of the property key, reverse it,
+    # and find the first 1
+    return bin(property_key)[::-1].index("1")
+
+
+async def async_bulk_set_partial_config_parameters(
+    node: Node,
+    property_: int,
+    new_value: Union[int, Dict[int, Union[int, str]]],
+) -> CommandStatus:
+    """Bulk set partial configuration values on this node."""
+    config_values = node.get_configuration_values()
+    property_values = [
+        value for value in config_values.values() if value.property_ == property_
+    ]
+
+    # If we can't find any values with this property, the property is wrong
+    if len(property_values) == 0:
+        raise NotFoundError(
+            f"Configuration parameter {property_} for node {node.node_id} not found"
+        )
+
+    # If we only find one value with this property_, we know this value isn't split
+    # into partial params
+    if len(property_values) == 1:
+        raise ValueTypeError(
+            f"Configuration parameter {property_} for node {node.node_id} does not "
+            "have partials"
+        )
+
+    # If new_value is a dictionary, we need to calculate the full value to send
+    if isinstance(new_value, dict):
+        temp_value = 0
```

Comment:
If you don't set **all** partials, this needs to be the currently cached (full) value for the config param.

Reply:
Right, we grab the missing partials values to add to the payload here: https://github.com/home-assistant-libs/zwave-js-server-python/pull/177/files#diff-f10e0f95086413d4ac47fadfd902b56d63ea3763366a8a8cb988cbdc7184a52cR132